### ISSUE TITLE
Cannot undo while text is selected

### ISSUE THREAD

**Issue by pdubroy** [tldraw/tldraw#1635]
### What happened?

**Repro steps:**

1. Add some text
2. Draw a shape
3. Select some of the text
4. Hit ⌘-Z a bunch of times.

**What happened:**
The shape drawn in step 2 never gets removed.

**What I expected:**
The shape should eventually get removed from the canvas. It works once I clear the text selection.

https://github.com/tldraw/tldraw/assets/17293/5353d949-6567-485d-9213-233496d39274

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by pdubroy** [comment]
fwiw I tried the same in a few other tools:

- Keynote (macOS): works as expected (shape gets removed)
- Notes (macOS): works as expected
- Figma: works as expected
- Excalidraw: same behaviour as tldraw

**Comment by steveruizok** [comment]
Thanks for the issue @pdubroy. Sorry if it's a major pain point but we'll need to keep this one as it is for now. I did a bit of research and unfortunately any implementation would mean managing the text clipboard and text clipboard state ourselves, which would be a major lift. Happy to re-open once we've got more capacity or if we go in another direction that forces our hand. 

If you're a future user who sees this as a paper cut, feel free to give a reply so we know.
